### PR TITLE
更换 HttpClient 为 OkHttp

### PR DESCRIPTION
**Pull request by paulzzh**
尝试修复 `selector manager closed`
变更挺多，需要仔细测试

## Release Notes

- **New Features**
  - Transitioned to using OkHttpClient for HTTP requests across multiple classes, enhancing performance and flexibility.
  - Introduced a new `ProgressListener` interface for tracking HTTP request progress.

- **Bug Fixes**
  - Improved error handling for HTTP requests and responses, ensuring more robust exception management.

- **Documentation**
  - Updated method signatures and removed deprecated HTTP client references to reflect the new implementation.

These changes collectively enhance the application's HTTP handling capabilities, providing a more efficient and reliable networking experience.

### CODE REVIEW

File: src/main/java/com/ghostchu/peerbanhelper/ipdb/IPDB.java [hunk 1]
```diff
@@ -280,20 +278,22 @@
...
+                        Request request = original.newBuilder()
+                                .header("User-Agent", userAgent)
+                                //.header("Accept-Encoding", "gzip,deflate")
+                                .header("Content-Type", "application/json")
+                                .method(original.method(), original.body())
+                                .build();
+                        return chain.proceed(request);
                     }
                 })
                 .build();
```
[the remark this answers is not in the document]
Comment (paulzzh) [reply src/main/java/com/ghostchu/peerbanhelper/ipdb/IPDB.java:299]: okhttp 默认 header 带 Accept-Encoding: gzip
覆写会出问题，需要自己处理被压缩过的body
如果不覆写，okhttp内部处理，后续拿到的是明文body

File: src/main/java/com/ghostchu/peerbanhelper/util/HTTPUtil.java [hunk 2]
```diff
@@ -79,113 +82,163 @@
...
         if (shouldRetry(resp, t, count)) {
             if (resp == null) {
-                log.warn("Request to {} failed, retry {}/{}: {}", request.uri().toString(), count, MAX_RESEND, t.getClass().getName() + ": " + t.getMessage());
+                log.warn("Request to {} failed, retry {}/{}: {}", request.url(), count, MAX_RESEND, t.getClass().getName() + ": " + t.getMessage());
             } else {
-                log.warn("Request to {} failed, retry {}/{}: {} ", request.uri().toString(), count, MAX_RESEND, resp.statusCode() + " - " + resp.body());
+                try {
+                    log.warn("Request to {} failed, retry {}/{}: {} ", request.url(), count, MAX_RESEND, resp.code() + " - " + resp.body().string());
+                } catch (IOException ignored) {
+                }
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/util/HTTPUtil.java:237]: 这个 try 块儿确实可以删了

**Comment by paulzzh** [comment]
@coderabbitai review

**Comment by paulzzh** [comment]
快端下去吧

**Pull request closed without merging**